### DISCUSSION TITLE
Investigate how to upload Tye2 schema to use schema globally

### DISCUSSION THREAD

**Discussion by drakon660** [tye-v2/tye2#41] [Ideas]
tye2 yaml can be used from VS Code with intellisense, but how we gonna approach this?

![image](https://github.com/tye-v2/tye2/assets/3002778/50fc945d-e6c9-48f7-b754-142629b331f5)

**Comment by razvangoga** [comment]
you can add it like this in the VSCode config - https://github.com/tye-v2/tye2/blob/main/src/schema/README.md

i've been using it for some time and it works ok (that how i found out that the schema was incomplete if you remember that fix you did to it about the routes)

there is also this https://github.com/SchemaStore/schemastore?tab=readme-ov-file / https://www.schemastore.org/json/ where the old tye seems to have been already registered

**Comment by drakon660** [comment]
Yes I remember, and I know that I can add it to vs code config, but the question is do we want to have our own? or leave it as it is?

**Reply by razvangoga** [reply]
this is a different app / tool than tye so yes, we should definitely have our own (and we already have a different schema thanks to your fix)

sticking to tye's is limiting - in the future we may want to add / change it

**Reply by drakon660** [reply]
so maybe add issue to that to not forget.

**Reply by razvangoga** [reply]
added it on the todo list of the v1 roadmap issue #2

**Comment by Mrxx99** [comment]
I think adding to schemastore makes the most sense, because local config has to be done manually and would only work that way in VSCode

**Reply by razvangoga** [reply]
agree in principle, but i'm not 100% clear on how the IDEs would disambiguate between tye and tye2 (based on the filename?)

**Reply by drakon660** [reply]
ask ms to free extension, is possible?

**Reply by razvangoga** [reply]
we cloud ask on the schemastore repo i guess

where did you get the screenshot from up top?

**Reply by drakon660** [reply]
orginal Tye repo I think, use github search.
